# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1440014243 14400
#      Wed Aug 19 15:57:23 2015 -0400
# Node ID 9267c95dbd71b10600fd11646a642dc49675d6b6
# Parent  b3f56ed8d1f9a789be77d3e94914185aa973f1be
avoid signed/unsigned comparison warning

* txt-eng-ft.cc (ft_render::process_character): Declare loop counters
as unsigned ints.

diff --git a/libinterp/corefcn/txt-eng-ft.cc b/libinterp/corefcn/txt-eng-ft.cc
--- a/libinterp/corefcn/txt-eng-ft.cc
+++ b/libinterp/corefcn/txt-eng-ft.cc
@@ -555,18 +555,18 @@ ft_render::process_character (FT_ULong c
 
                   // 'w' seems to have a negative -1
                   // face->glyph->bitmap_left, this is so we don't
                   // index out of bound, and assumes we we allocated
                   // the right amount of horizontal space in the bbox.
                   if (x0 < 0)
                     x0 = 0;
 
-                  for (int r = 0; r < bitmap.rows; r++)
-                    for (int c = 0; c < bitmap.width; c++)
+                  for (unsigned int r = 0; r < bitmap.rows; r++)
+                    for (unsigned int c = 0; c < bitmap.width; c++)
                       {
                         unsigned char pix = bitmap.buffer[r*bitmap.width+c];
                         if (x0+c < 0 || x0+c >= pixels.dim2 ()
                             || y0-r < 0 || y0-r >= pixels.dim3 ())
                           {
                             //::warning ("ft_render: pixel out of bound (char=%d, (x,y)=(%d,%d), (w,h)=(%d,%d)",
                             //           str[i], x0+c, y0-r, pixels.dim2 (), pixels.dim3 ());
                           }
